### ISSUE TITLE
[ExplicitColumns] and [Reference(ReferenceType.Foreign,] bugs

### ISSUE THREAD

**Issue by euioa** [schotime/NPoco#315]
Hi,
Assume theses mapped classes : 

```
[TableName("Users")]
[ExplicitColumns]
[PrimaryKey("User_ID", AutoIncrement = true)]
public class User
{
    [Column("User_ID")]
    public int ID { get; set; }

    [Column("Name")]
    public string Name
    { get; set; }

    [Reference(ReferenceType.Foreign, ColumnName = "UserGroup_ID", ReferenceMemberName = "ID")]
    public UserGroup UserGroup { get; set; }
}


[TableName("UserGroups")]
[ExplicitColumns]
[PrimaryKey("User_ID", AutoIncrement = true)]
public class UserGroup
{
    [Column("UserGroup_ID")]
    public int ID
    {get;set;}

    [Column("UserGroup")]
    public string Name
    { get;set; }

    [Reference(ReferenceType.Many, ColumnName = "UserGroup_ID", ReferenceMemberName = "UserGroupID")]
    [Column("Users")]
    public List<User> Users
    {
        get;set;
    }
}
```

Tests : 

```
var a = db.Query<User>().ToList();  // works. ok...
var a= db.Query<User>().Include(o => o.UserGroup).ToList(); // Throw InvalidOperationException. If I remove [ExplicitColumns] then it works.
```

Now, I add this property to User class

```
    [Column("UserGroup_ID")]
    public string UserGroupID
    { get; set; }
```

Tests : 

`var a = db.Query<User>().ToList(); // this works, but if i remove [ExplicitColumns] it throw ArgumentException (duplicate key)`

**Comment by schotime** [comment]
What if you remove [Column("users")] seeing as there is no column "users"?

**Comment by euioa** [comment]
same after [Column("users")remove

**Comment by schotime** [comment]
Will need a small app that can be run standalone to reproduce this.

**Comment by euioa** [comment]
[ConsoleApplication1.zip](https://github.com/schotime/NPoco/files/434605/ConsoleApplication1.zip)

hope this small app is ok for you ?

**Comment by schotime** [comment]
Hey, Perfect.
It was a bug and its been fixed in 3.4.2.
I have uploaded your sample application with the configs all correct and working too.
Cheers.

[ConsoleApplication1-fixed.zip](https://github.com/schotime/NPoco/files/448816/ConsoleApplication1-fixed.zip)
